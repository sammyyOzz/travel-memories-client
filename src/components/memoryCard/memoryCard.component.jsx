import React from 'react';
import * as Styles from './memoryCard.styles'
import { baseUrl } from '../../utils/services/axios'
import { useDispatch } from 'react-redux';
import { setMemoryForDisplay } from '../../redux/memories/memories.slice';
import PropTypes from 'prop-types'



export function MemoryCard({ _id, imageUrl, place, description, user: { name } }) {
    
    const dispatch = useDispatch()

    const _setMemoryForDisplay = (data) => dispatch(setMemoryForDisplay(data))


    const handleClick = () => {
        _setMemoryForDisplay({ _id, imageUrl, place, description, name })
    }

    return (
        <Styles.Root onClick={handleClick}>
            <Styles.ImageContainer>
                <Styles.Image src={`${baseUrl}/images/${imageUrl}`} alt="" />
            </Styles.ImageContainer>
            <Styles.Place>{ place }</Styles.Place>
            <Styles.Description>{ description }</Styles.Description>
            <Styles.Footer>{ `--${name}--` }</Styles.Footer>
        </Styles.Root>
    )
}

MemoryCard.defaultProps = {
    _id: `${Math.random()}`,
    user: {
        name: PropTypes.string
    }
}

MemoryCard.propTypes = {
    _id: PropTypes.string.isRequired,
    imageUrl: PropTypes.string,
    place: PropTypes.string,
    description: PropTypes.string,
    user: PropTypes.object
}