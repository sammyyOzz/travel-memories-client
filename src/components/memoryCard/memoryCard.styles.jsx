import styled from 'styled-components'

export const Root = styled.div`
    box-shadow: 0 2px 6px skyblue;
    border-radius: 20px;
    cursor: pointer;
    background-color: #0c0d1b;
    position: relative;

    &:hover img { 
        transform: scale(1.5);
        transition: 1000ms linear all;
    }
`

export const ImageContainer = styled.div`
    overflow: hidden;
    height: 250px;
    width: 100%;
    border-radius: 20px 20px 0 0;
`

export const Image = styled.img`
    height: 100%;
    width: 100%;
    object-fit: cover;
    transition: 1000ms linear all;
`

export const Place = styled.h2`
    margin: 20px;
    color: white;
`

export const Description = styled.div`
    padding: 10px 20px 40px 20px;
    box-sizing: border-box;
    border-radius: 0 0 20px 20px;
    color: white;
`

export const Footer = styled.div`
    font-size: 14px;
    color: skyblue;
    /* margin: 10px 20px; */
    position: absolute;
    bottom: 10px;
    left: 20px;
`