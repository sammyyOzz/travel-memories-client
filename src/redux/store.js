import { configureStore } from '@reduxjs/toolkit'
import memoriesReducer from './memories/memories.slice'
import authReducer from './auth/auth.slice'


export default configureStore({
    reducer: {
      auth: authReducer,
      memories: memoriesReducer,
    },
})
  