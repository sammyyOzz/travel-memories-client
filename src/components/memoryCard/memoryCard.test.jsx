import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { MemoryCard } from "./memoryCard.component"
import store from '../../redux/store'

describe('<MemoryCard />', () => {
    test('passes snapshot', () => {
        expect(render(
            <Provider store={store}>
                <MemoryCard />
            </Provider>
        )).toMatchSnapshot()
    })
})