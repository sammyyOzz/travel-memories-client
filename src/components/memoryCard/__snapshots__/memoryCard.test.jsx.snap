// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoryCard /> passes snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ iUIcWA"
      >
        <div
          class="sc-gsnTZi eYiPVv"
        >
          <img
            alt=""
            class="sc-dkzDqf esSDri"
            src="http://localhost:5000/images/undefined"
          />
        </div>
        <h2
          class="sc-hKMtZM fUXOsT"
        />
        <div
          class="sc-eCYdqJ fDmgfH"
        />
        <div
          class="sc-jSMfEi hMaAOr"
        >
          --function () { [native code] }--
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ iUIcWA"
    >
      <div
        class="sc-gsnTZi eYiPVv"
      >
        <img
          alt=""
          class="sc-dkzDqf esSDri"
          src="http://localhost:5000/images/undefined"
        />
      </div>
      <h2
        class="sc-hKMtZM fUXOsT"
      />
      <div
        class="sc-eCYdqJ fDmgfH"
      />
      <div
        class="sc-jSMfEi hMaAOr"
      >
        --function () { [native code] }--
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
